bot.py

```diff
@@ -74,26 +76,30 @@ def _setup_api(self):
         self.api.get_player()
 
         response_dict = self.api.call()
-        #print('Response dictionary: \n\r{}'.format(json.dumps(response_dict, indent=2)))
-        currency_1="0"
-        currency_2="0"
-        if 'amount' in response_dict['responses']['GET_PLAYER']['profile']['currency'][0]:
-            currency_1=response_dict['responses']['GET_PLAYER']['profile']['currency'][0]['amount']
-        if 'amount' in response_dict['responses']['GET_PLAYER']['profile']['currency'][1]:
-            currency_2=response_dict['responses']['GET_PLAYER']['profile']['currency'][1]['amount']
-        print 'Profile:'
-        print '    Username: ' + str(response_dict['responses']['GET_PLAYER']['profile']['username'])
-        print '    Bag size: ' + str(response_dict['responses']['GET_PLAYER']['profile']['item_storage'])
-        print '    Pokemon Storage Size: ' + str(response_dict['responses']['GET_PLAYER']['profile']['poke_storage'])
-        print '    Account Creation: ' + str(response_dict['responses']['GET_PLAYER']['profile']['creation_time'])
-        print '    Currency: '
-        print '        ' + str(response_dict['responses']['GET_PLAYER']['profile']['currency'][0]['type']) + ': ' + str(currency_1)
-        print '        ' + str(response_dict['responses']['GET_PLAYER']['profile']['currency'][1]['type']) + ': ' + str(currency_2)
+        player_profile = response_dict['responses']['GET_PLAYER']['profile']
+
+        for currency in player_profile.get('currency', []):
+            player_profile[currency['type']] = currency.get('amount', 0)
+
+        self.log.info('#' * 30)
+        self.log.info('Profile:')
+        self.log.info('Username: {username}'.format(**player_profile))
+        self.log.info('Bag Size: {item_storage}'.format(**player_profile))
+        self.log.info('Pokemon Storage: {poke_storage}'.format(**player_profile))
+        self.log.info('Account Creation: {creation_time}'.format(**player_profile))
+        self.log.info('Pokecoin: {POKECOIN}'.format(**player_profile))
+        self.log.info('Stardust: {STARDUST}'.format(**player_profile))
+        self.log.info('#' * 30)
```

Comment:
@berend great refactoring with `str.format`, awesome! I didn't know this was possible.